avocado/plugins/runners/vmimage.py

```diff
@@ -0,0 +1,164 @@
+import multiprocessing
+import signal
+import sys
+import time
+import traceback
+from multiprocessing import set_start_method
+
+from avocado.core.exceptions import TestInterrupt
+from avocado.core.nrunner.app import BaseRunnerApp
+from avocado.core.nrunner.runner import RUNNER_RUN_CHECK_INTERVAL, BaseRunner
+from avocado.core.utils import messages
+from avocado.core.utils.messages import start_logging
+from avocado.plugins.vmimage import download_image
+from avocado.utils import vmimage
+
+
+class VMImageRunner(BaseRunner):
+    """
+    Runner for dependencies of type vmimage.
+    This runner uses the vmimage plugin's download_image function which handles:
+    1. Checking if the image exists in cache
+    2. Downloading the image if not in cache
+    3. Storing the image in the configured cache directory
+    4. Returning the cached image path
+    """
+
+    name = "vmimage"
+    description = "Runner for dependencies of type vmimage"
+
+    @staticmethod
+    def signal_handler(signum, frame):  # pylint: disable=W0613
+        if signum == signal.SIGTERM.value:
+            raise TestInterrupt("VM image operation interrupted: Timeout reached")
+
+    @staticmethod
+    def _run_vmimage_operation(runnable, queue):
+        try:
+            signal.signal(signal.SIGTERM, VMImageRunner.signal_handler)
+            start_logging(runnable.config, queue)
+            # Get parameters from runnable.kwargs
+            provider = runnable.kwargs.get("provider")
+            version = runnable.kwargs.get("version")
+            arch = runnable.kwargs.get("arch")
+
+            if not provider:
+                stderr = "Missing required parameter: provider"
+                queue.put(messages.StderrMessage.get(stderr.encode()))
+                queue.put(messages.FinishedMessage.get("error"))
+                return
+
+            message = f"Getting VM image for {provider}"
+            if version:
+                message += f" {version}"
+            if arch:
+                message += f" {arch}"
+            queue.put(messages.StdoutMessage.get(message.encode()))
+
+            # download_image will use cache if available, otherwise download
+            provider_normalized = provider.lower()
```

Comment:
The normalization is not needed since vmimage utility takes care of it. See https://github.com/avocado-framework/avocado/blob/master/avocado/utils/vmimage.py#L730